File: Quiz 1/Palak_Rastogi_590018052/Quiz_1_Debugging_Question_2.py
'''
Bug : variable max is not initialized due to which max 
will start with a garbage value and none of the array element will replace mmax_value
'''

def find_max(arr):
    max_val=arr[0]; #Initialize with the first element

    for i in range (1,len(arr)):
        if(arr[i] > max_val):
            max_val = arr[i]
    return max_val

#Example
arr=[5,7,2,9,14,8]
print('Maximum element is: ', find_max(arr))

File: Quiz_1/Prabhjeet_590013286/Quiz_1_Debugging_Question_2.c
int findMax(int arr[], int n)
{
    if (n <= 0) return -1;

    int max = arr[0];

    for (int i = 1; i < n; i++)
    {
        if (arr[i] > max)
        {
            max = arr[i];
        }
    }

    return max;
}
